smithy-linters/src/main/java/software/amazon/smithy/linters/MissingSensitiveTraitValidator.java

```diff
@@ -0,0 +1,196 @@
+import software.amazon.smithy.utils.ListUtils;
+import software.amazon.smithy.utils.SetUtils;
+
+/**
+ * <p>Validates that shapes and members that possibly contain sensitive data are marked with the sensitive trait.
+ */
+public final class MissingSensitiveTraitValidator extends AbstractValidator {
+    public static final String SENSITIVE = "sensitive";
+    static final Set<String> DEFAULT_SENSITIVE_WORDS = SetUtils.of(
+            "authorization",
+            "bank",
+            "billing",
+            "birth",
+            "credential",
+            "email",
+            "ethnicity",
+            "insurance",
+            "license",
+            "passphrase",
+            "passport",
+            "password",
+            "phone",
+            "private",
+            "secret",
+            "sensitive",
+            "telephone",
+            "token",
+            "username",
+            "zip"
+    );
+    // "Phrases" will be searched for without regard to word boundaries. (This is so that user configuration )
+    static final Set<String> DEFAULT_SENSITIVE_PHRASES = SetUtils.of(
+            "accesskey",
+            "creditcard",
+            "firstname",
+            "lastname",
+            "plaintext",
+            "taxpayer"
+    );
+
+    public static final class Provider extends ValidatorService.Provider {
+        public Provider() {
+            super(MissingSensitiveTraitValidator.class, node -> {
+                NodeMapper mapper = new NodeMapper();
+                return new MissingSensitiveTraitValidator(
+                        mapper.deserialize(node, MissingSensitiveTraitValidator.Config.class));
+            });
+        }
+    }
+
+    /**
+     * MissingSensitiveTrait configuration.
+     */
+    public static final class Config {
+        private List<String> phrases = ListUtils.of();
+        private boolean excludeDefaults;
+
+        public List<String> getPhrases() {
+            return phrases;
+        }
+
+        public void setPhrases(List<String> phrases) {
+            this.phrases = phrases;
+        }
+
+        public boolean getExcludeDefaults() {
+            return excludeDefaults;
+        }
+
+        public void setExcludeDefaults(boolean excludeDefaults) {
+            this.excludeDefaults = excludeDefaults;
+        }
+    }
+
+    private final Set<String> sensitiveWords;
+    private final Set<String> sensitivePhrases;
+
+    private MissingSensitiveTraitValidator(Config config) {
+        if (!config.getExcludeDefaults()) {
+            Set<String> phrasesInit = new HashSet<>(DEFAULT_SENSITIVE_PHRASES);
+            phrasesInit.addAll(config.getPhrases()
+                    .stream()
+                    .map(phrase -> phrase.toLowerCase(Locale.US))
+                    .collect(Collectors.toSet()));
+            sensitivePhrases = Collections.unmodifiableSet(phrasesInit);
+            sensitiveWords = Collections.unmodifiableSet(DEFAULT_SENSITIVE_WORDS);
```

Comment:
Since the docs aren't finished above, I'm not sure why these need to be separate. Should this be configurable?

Reply:
Made both configurable separately and added details in the spec about when to use each one